### DISCUSSION TITLE
Sudden appearance and challenging debugging of `TypeError: Cannot read properties of undefined ...`

### DISCUSSION THREAD

**Discussion by charlieforward9** [vitest-dev/vitest#5059] [General]
I am able to run my project without issue, but when I try to run Vitest, I get the daunting `TypeError: Cannot read properties of undefined ...` on certain files of the codebase that do not have anything to do with the test being run, which ends up making the test fail.

The strange thing is that a few months ago when I put these tests together, this was not happening. Obviously I have made changes to the codebase, but the tests have not changed at all over this time, and the dependencies of the particular tests remain the same as they were when I first built and successfully ran them.

 Could I please be pointed to resources to understand why this is happening and how to work around or solve it? 
 
 Thank you

**Comment by charlieforward9** [comment]
After much debugging, I am realizing that Vitest does not handle Barrel Files very well. I have had to modify many of my import statements across my codebase to account for the tests that depend on modules that depend on the barrel file. 

For additional context, I am using the `jsdom` environment. Not sure if that has something to do with it.

It would be nice to learn WHY this is the case, and know if there are plans to work on this in the future to make Vitest more "barrel file" friendly. Consistent with the way the dev app runs with Vite.

**Comment by hi-ogawa** [comment]
If Vitest behaves inconsistently due to the use barrel files, then that would be a bug, but it's difficult to see what the actual issue is from your description. We would appreciate if you can provide a reproduction.

Maybe these two things could be potentially related:
- do you use monorepo? (e.g. tests in one package is depending on other package in monorepo?)
- do you use `vi.mock`?

**Reply by charlieforward9** [reply]
I am using a single repository to test the app, but have not *package*ified the application. The folder structure is currently designed to be very modular and eventually able to be turned into unique pacakges, but there is only on `package.json` in the repo, therefore I dont think the answer to the first question is yes. 

I do use vi.mock.

I will look into creating a reproduction soon as I would prefer to run all my imports from the barrelfile vs individual modules. Will get to this soon.
